src/nvim/lua/xdiff.c

```diff
@@ -61,20 +66,90 @@ static int hunk_locations_cb(long start_a, long count_a, long start_b, long coun
   if (count_b > 0) {
     start_b += 1;
   }
-
   lua_State *lstate = (lua_State *)cb_data;
-  lua_createtable(lstate, 0, 0);
+  if (linematch) {
+    const char **diff_begin = xmalloc(sizeof(char *) * 2);
```

Comment:
`diff_begin` is never freed, however you should be able to stack allocate it instead.

```suggestion
    const char *diff_begin[2];
```

Reply:
it's now allocated on stack